pyxrf/model/scan_metadata.py

```diff
@@ -0,0 +1,213 @@
+        #   In principle the descriptions may not be set at all.
+        self.descriptions = self._gen_default_descriptions()
+
+        self._print_order = self._gen_default_print_order()
+
+    def _gen_default_print_order(self):
+        """
+        This function must have meaningful implementation in the child class
+        """
+        return []
+
+    def _gen_default_descriptions(self):
+        """
+        This function must have meaningful implementation in the child class
+        """
+        return {}
+
+    # The following methods override standard operators to enable direct access
+    #   to ``self.values`` dictionary
+
+    def __getitem__(self, key):
+        return self._values[key]
+
+    def __setitem__(self, key, value):
+        self._values[key] = value
+
+    def __delitem__(self, key):
+        del self._values[key]
+
+    def __contains__(self, key):
+        return key in self._values
+
+    def __iter__(self):
+        return self._values.__iter__()
+
+    def keys(self):
+        return self._values.keys()
+
+    def values(self):
+        return self._values.values()
+
+    def items(self):
+        return self._values.items()
+
+    def get_metadata_dictionary(self):
+        """
+        Returns reference to metadata dictionary ``self._values``
+        """
+        return self._values
+
+    def is_metadata_available(self):
+        """
+        Returns True if at least one metadata key:value pair is available.
+        Otherwise returns False.
+        """
+        return bool(self._values)
+
+    def get_formatted_output(self):
+        """
+        Returns formatted metadata in the form ready for printing.
+        Formatting is performed based on specifications located in ``self._print_order``
+        list. Key names are replaced by descriptions from ``self.descriptions`` if
+        the available.
+        """
+        str_out = ""
+
+        printed_keys = set()
+        flag_empty_line = False  # Indicates if empty line was just printed
+
+        for ptn in self._gen_default_print_order():
+            # We don't want to print multiple empty lines in a row
+            if ptn == "" and not flag_empty_line:
+                str_out += "\n"
+                flag_empty_line = True
+                continue
+
+            for key, v in self._values.items():
+                if re.search(f"^{ptn}$", key) and (key not in printed_keys):
+                    flag_empty_line = False  # Something is getting printed
+                    # Mark the entry as printed
+                    printed_keys.add(key)
+                    # Extracted printable expression for the key
+                    s_key = key
+                    if key in self.descriptions and self.descriptions[key]:
+                        s_key = self.descriptions[key]
+                        # Capitalize the first letter
+                        s_key = s_key[:1].upper() + s_key[1:]
```

Comment:
Fixed. I don't think that the alternatives are available, since this is custom output. And it is not supposed to be machine-readable.

Reply:
You can check https://docs.python.org/3/library/textwrap.html?highlight=textwrap and the backport project https://github.com/jonathaneunice/textwrap3 (which we package). I think it may be what you need.

https://pypi.org/project/ansiwrap/ is another option.

Reply:
I went ahead and tried it myself:
```py
In [5]: long_line = 'Wraps the single paragraph in text (a string) so every line is at most width characters long. Returns a list of output
   ...: lines, without final newlines.'
In [27]: print(textwrap.fill(long_line, width=40, initial_indent='  ', subsequent_indent='  '))
  Wraps the single paragraph in text (a
  string) so every line is at most width
  characters long. Returns a list of
  output lines, without final newlines.
```

Reply:
Implemented using textwrap. But it is still not a one-liner.

Reply:
Cool, thanks! It does not have to be a one-liner anyway. My point was to use already available tools if they can do what you want.